src/System.Runtime/tests/System/String.cs

```diff
@@ -1378,33 +1324,25 @@ public static void TestStartsWithInvalid()
         Assert.Throws<ArgumentNullException>(() => s.StartsWith(null, StringComparison.OrdinalIgnoreCase));
     }
 
-    [Fact]
-    public static void TestSubstring()
+    [Theory]
+    [InlineData("Hello", 2, null, "llo")] //null = don't use this overload
+    [InlineData("Hello", 0, null, "Hello")]
+    [InlineData("Hello", 5, null, "")]
```

Comment:
Rather than having the above three cases, you could instead have the body of the test be:
```C#
if (startIndex + length == text.Length)
{
    Assert.Equal(expected, text.Substring(startIndex);
}
Assert.Equal(expected, text.Substring(startIndex, length);
```
and your existing InlineData's will cover the same cases.